src/main/java/com/ss/training/utopia/orchestrator/security/UserDetailsClass.java

```diff
@@ -0,0 +1,67 @@
+package com.ss.training.utopia.orchestrator.security;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+/**
+ * @author Trevor Huis in 't Veld
+ * @author Justin O'Brien
+ */
+public class UserDetailsClass implements UserDetails {
+
+	private static final long serialVersionUID = 1557339291337891604L;
+	private String username;
+	private String password;
+	private boolean enabled;
+	private List<GrantedAuthority> authorities;
+
+	public UserDetailsClass(User user) {
+		this.username = user.getUsername();
+		this.password = user.getPassword();
+		this.enabled = user.isEnabled();
+		this.authorities = Arrays.stream(user.getRole().split(",")).map(role -> "ROLE_" + role)
+				.map(SimpleGrantedAuthority::new).collect(Collectors.toList());
```

Comment:
Something that didn't occur to me before is that this part can be made simpler now that each user only has one role. This code splits up a comma-separated string to make a list of all the roles. Now all we need to do is just make a one-element list with the role (which of course is what this will do when there's only one role).